src/index.ts

```diff
@@ -261,39 +287,29 @@ export class EmailLinkStrategy<User> extends Strategy<
 
     try {
       // If we get here, the user clicked on the magic link inside email
-      const magicLink = session.get(this.sessionMagicLinkKey) ?? ''
+      const magicLink = session.get('magicLink') ?? ''
       const { emailAddress: email, form } = await this.validateMagicLink(
         request.url,
-        await this.decrypt(magicLink)
+        await this.decrypt(magicLink as string)
       )
       // now that we have the user email we can call verify to get the user
       user = await this.verify({ email, form, magicLinkVerify: true })
     } catch (error) {
-      // if something happens, we should redirect to the failureRedirect
-      // and flash the error message, or just throw the error if failureRedirect
-      // is not defined
-      if (!options.failureRedirect) {
-        throw error
-      }
       const { message } = error as Error
-      session.flash(this.sessionErrorKey, { message })
-      const cookie = await sessionStorage.commitSession(session)
-      throw redirect(options.failureRedirect, {
+      session.flash('error', message)
+      const cookie = await this.sessionStorage.commitSession(session)
+      throw redirect(this.failureRedirect, {
         headers: { 'Set-Cookie': cookie },
       })
     }
 
-    if (!options.successRedirect) {
-      return user
-    }
-
     // remove the magic link and email from the session
-    session.unset(this.sessionMagicLinkKey)
-    session.unset(this.sessionEmailKey)
+    session.unset('magicLink')
```

Comment:
couple of observations here:
1. it looks like the ability to set one's own keys for the cookie storage is no longer available? This feels like a regression in my opinion
2. to minimize breaking changes we should leave the default values as is e.g. auth:magiclink

Reply:
Still struggling with this one. I just refactored my app to use this set of changes and I had to refactor the shape of my session object. It feels kind of heavy-handed for this auth Strategy to strongly dictate the shape of the Session data